packages/office-ui-fabric-react/src/components/SearchBox/SearchBox.base.tsx

```diff
@@ -0,0 +1,206 @@
+import * as React from 'react';
+import { ISearchBoxProps, ISearchBoxStyleProps, ISearchBoxStyles } from './SearchBox.types';
+import {
+  BaseComponent,
+  autobind,
+  css,
+  getId,
+  KeyCodes,
+  customizable,
+  classNamesFunction
+} from '../../Utilities';
+
+import { IconButton } from '../../Button';
+import { Icon } from '../../Icon';
+
+import { FocusZone } from '../../FocusZone';
+
+const getClassNames = classNamesFunction<ISearchBoxStyleProps, ISearchBoxStyles>();
+
+
+export interface ISearchBoxState {
+  value?: string;
+  hasFocus?: boolean;
+  id?: string;
+}
+
+@customizable('SearchBox', ['theme'])
+export class SearchBoxBase extends BaseComponent<ISearchBoxProps, ISearchBoxState> {
+  public static defaultProps: ISearchBoxProps = {
+    labelText: 'Search',
+  };
+
+  private _rootElement: HTMLElement;
+  private _inputElement: HTMLInputElement;
+  private _latestValue: string;
+
+  public constructor(props: ISearchBoxProps) {
+    super(props);
+
+    this.state = {
+      value: props.value || '',
+      hasFocus: false,
+      id: getId('SearchBox')
+    };
+  }
+
+  public componentWillReceiveProps(newProps: ISearchBoxProps) {
+    if (newProps.value !== undefined) {
+      this._latestValue = newProps.value;
+      this.setState({
+        value: newProps.value
+      });
+    }
+  }
+
+  public render() {
+    let { labelText, className, disabled, underlined, getStyles, theme } = this.props;
+    let { value, hasFocus, id } = this.state;
+
+    const classNames = getClassNames(getStyles!, { theme: theme!, underlined, hasFocus, disabled });
+
+    return (
+      <div
+        ref={ this._resolveRef('_rootElement') }
+        className={ css(className, classNames.root,
```

Comment:
Shouldn't we add most of these classNames in the .styles file? I'd like to see us get rid of the css() function calls as a part of this conversion

Reply:
as a prop, classname has a contract. pass it in and it shows up in the component markup. Moving that into the style file means that if you create a custom searchbox, you are responsible for applying that classname. Not sure if that is a better experience.

Reply:
I dont believe this is how we've done it for other components.

I also don't expect that using a classname and providing  a get styles override is a common scenario is it?